haystack/nodes/prompt/invocation_layer/amazon_bedrock.py

```diff
@@ -0,0 +1,368 @@
+from abc import ABC, abstractmethod
+import json
+import logging
+import re
+from typing import Any, Optional, Dict, Type, Union, List
+
+from haystack.errors import AWSConfigurationError, AmazonBedrockConfigurationError, AmazonBedrockInferenceError
+from haystack.lazy_imports import LazyImport
+from haystack.nodes.prompt.invocation_layer.aws_base import AWSBaseInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import (
+    DefaultPromptHandler,
+    DefaultTokenStreamingHandler,
+    TokenStreamingHandler,
+)
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install farm-haystack[aws]'") as boto3_import:
+    from botocore.exceptions import ClientError
+
+
+class BedrockModelAdapter(ABC):
+    """
+    Base class for Amazon Bedrock model adapters.
+    """
+
+    def __init__(self, model_kwargs: Dict[str, Any], max_length: Optional[int]) -> None:
+        self.model_kwargs = model_kwargs
+        self.max_length = max_length
+
+    @abstractmethod
+    def prepare_body(self, prompt: str, **inference_kwargs) -> Dict[str, Any]:
+        """Prepares the body for the Amazon Bedrock request."""
```

Comment:
is it clear what "body"? (it's not clear to me :) )

Reply:
Yes, it's really technical. But we are basically preparing the `body` param of the beckrock call. Body is sometimes used as synonym for the payload of a request.